docs/src/tutorial.jl

```diff
@@ -708,161 +707,81 @@ end
 # and, when making the model we would specify
 
 model = StandardABM(
-    Union{SchellingAgent, Politician}, # type of agents
+    Union{Schelling, Politician}, # type of agents
     space; # space they live in
 )
 
 # Naturally, we would have to define a new agent stepping function that would
 # act differently depending on the agent type. This could be done by making
 # a function that calls other functions depending on the type, such as
 
-function union_step!(agent, model)
-    if typeof(agent) <: AgentSchelling
-        schelling_step!(agent, model)
-    elseif typeof(agent) <: Politician
-        politician_step!(agent, model)
-    end
+function agent_step!(agent::Schelling, model)
+    ## stuff.
+end
+
+function agent_step!(agent::Politician, model)
+    ## other stuff.
 end
 
 # and then passing
 
 model = StandardABM(
-    Union{SchellingAgent, Politician}, # type of agents
+    Union{Schelling, Politician}, # type of agents
     space; # space they live in
-    agent_step! = union_step!
+    agent_step!
 )
 
-# This approach also works with the [`@multiagent`](@ref) possibility we discuss below.
-# `Union` types however also offer the unique possibility of utilizing fully the Julia's
-# [multiple dispatch system](https://docs.julialang.org/en/v1/manual/methods/).
-# Hence, we can use the same function name and add dispatch to it, such as:
-
-function dispatch_step!(agent::SchellingAgent, model)
-    ## stuff.
-end
-
-function dispatch_step!(agent::Politician, model)
-    ## other stuff.
-end
-
-# and give `dispatch_step!` to the `agent_step!` keyword during model creation.
-
 # ## Multiple agent types with `@multiagent`
 
-# [`@multiagent`](@ref) is a macro, and hence not "native Julia" syntax like `Union`,
-# however it has been designed to be as similar to [`@agent`](@ref) as possible.
-# The syntax to use it is like so:
-
-@multiagent struct MultiSchelling{X}(GridAgent{2})
-    @subagent struct Civilian # can't re-define existing `Schelling` name
-        mood::Bool = false
-        group::Int
-    end
-    @subagent struct Governor{X} # can't redefine existing `Politician` name
-        group::Int
-        influence::X
-    end
-end
-
-# This macro created three names into scope:
-
-(MultiSchelling, Civilian, Governor)
-
-# however, only one of these names is an actual Julia type:
+# By using `@multiagent` it is often possible to improve the 
+# computational performance of simulations requiring multiple types,
+# while almost everything works the same 
 
-fieldnames(MultiSchelling)
+@multiagent MultiSchelling(Schelling, Politician) <: AbstractAgent
 
-# that contains all fields of all subtypes without duplication, while
+# Now you can create instances with
 
-fieldnames(Civilian)
+p = MultiSchelling'.Politician(model; pos = random_position(model), preferred_demographic = 1)
```

Comment:
I noticed this before and I thought it was a typo. What's the point of the `'` here? That in linear algebra stands for the transpose operator? Why do we have `MultiSchelling'.Politician` instead of `MultiSchelling.Politician`?

Reply:
I think you can overload `getfield` to work withe subtypes properly without the `'`. I admit I don't really understand the syntax here, what does `'` achieve and why it is here.

Reply:
Actually, can we just use `Politician` directly? Why do we have to preface it with `MultiAgent` at all? That would be te best.

Reply:
well it's actually a trick, overloading `getproperty(::Type{...})` is actually a mistake because it created problems, see https://github.com/JuliaLang/julia/discussions/48883#discussioncomment-5362804

Reply:
also, we need to preface it because `Politician` is already taken from the type itself

Reply:
Can we perhaps make an actual function instead of this funny syntax? If you think about it it is really atypical to write `Someting'.something`. My guess is that using the "transpose" operator on `MultiAgent` transforms it to something else, from which you can then access the field `Politician`. 

For a front end user, this may fly over their heads. I can easily imagine users forgetting `'` and then getting errors (that may or may not be informative). It is very unfortunate that we can't use direct dot `.` syntax...

Can we briefly brainstorm alternatives that are more transparent? Or, better yet, how does the `add_agent!` version work for MultiAgent models? do you give `MultiAgent'.Politician` as first argument and that's it? Perhaps we could provide an alternative syntax at the level of `add_agent!`? 

I don't think we should be promoting making agents outside of `add_agent!`. We have already "forbidden" the user to set the ID, so I don't think making agents on their own is even compatible with the publically declared api.

Reply:
I also read https://github.com/JuliaLang/julia/discussions/48883#discussioncomment-5362804 but I don't understand the issue/problem as we have a macro that creates `MultiAgent` so in essence we can make the fields of this object whatever we want. can you perhaps explain in layman terms why `MultiAgent.Politician` isn't possible?

Reply:
> do you give `MultiAgent'.Politician` as first argument and that's it? Perhaps we could provide an alternative syntax at the level of `add_agent!`?

yes, that is how it works. I don't have good ideas to make it work better unfortunately.

I think we could provide also a function (which would return a closure) but this is not that required here and I'm not sure how I would call it. But this is in my opinion not a blocking aspect for this PR, this could be always improved in a new PR, the one based on adjoint is here to stay anyway because it is backed by the underlying package

Reply:
the issue is from what I get that for any type by default you have

```
julia> struct A end

julia> propertynames(A)
(:name, :super, :parameters, :types, :instance, :layout, :hash, :flags)
```

but if you overload `getproperty(::Type, ::Symbol)` for (only) your type you practically creates problems because I think Julia invalidates a lot of unrelated code because of this. I don't understand the reason either but that is true as attested here https://github.com/Roger-luo/Expronicon.jl/issues/77 (final comment of the thread I mentioned)

Reply:
Can you write a simple function `gettype(MultiAgent, Type) = MultiAgent'.Type` and use this function in the tutorial? Also adding a sentence saying "We can't use `Politician` directly, as this is an existing type that doesn't have any refenrce to `MultiAgent`. We have to extract an incorporated `Politician` subtype, which we can do with `gettype`."

I am happy to discuss better names for `gettype` :D

Reply:
> but if you overload getproperty(::Type, ::Symbol) for (only) your type you practically creates problems 

We have been doing this for many many years with `AgentBasedModel` though: https://github.com/JuliaDynamics/Agents.jl/blob/main/src/core/model_abstract.jl#L116-L132  . And I've never heard of these problems. Besides, teh Julia manual itself says that `getproperty` is the function you overload for the `.` syntax. So this is a point I don't understand.

Reply:
In `AgentBasedModel` is different, there you don't have a type, there you have an instance of a type (`ABM` instead of `Type{ABM}`), that is okay and very encouraged actually I think

Reply:
that would be okay, maybe `get_type`? I think it's the tersest we can get probably...maybe `getconstr`? let me know your preference and we will go with that

Reply:
> In `AgentBasedModel` is different, there you don't have a type, there you have an instance of a type (`ABM` instead of `Type{ABM}`), that is okay and very encouraged actually I think

ooooooh yes of course, completely missed that... `Type{}` changes everything.

Reply:
Let's go with just `constructor`? We don't need `get`, since the function is named after what it returns.

Reply:
that's good for me, I have implemented it